in_toto/verifylib.py

```diff
@@ -943,35 +960,47 @@ def verify_threshold_equality(layout, chain_link_dict):
   for step in layout.steps:
     # Skip steps that don't require multiple functionaries
     if step.threshold <= 1:
+      log.info("Skipping threshold verification for step '{0}' with"
+          " threshold '{1}'...".format(step.name, step.threshold))
       continue
 
+    log.info("Verifying threshold for step '{0}' with"
+        " threshold '{1}'...".format(step.name, step.threshold))
     # Extract the key_link_dict for this step from the passed chain_link_dict
     key_link_dict = chain_link_dict[step.name]
 
     # Check if we have at least <threshold> links for this step
     if len(key_link_dict) < step.threshold:
-      raise ThresholdVerificationError("Step not performed"
-          " by enough functionaries!")
+      raise ThresholdVerificationError("Step '{0}' not performed"
+          " by enough functionaries!".format(step.name))
 
-    # take one exemplary link (e.g. the first in the step_link_dict)
-    compare_link = key_link_dict.values()[0]
+    # Take a reference link (e.g. the first in the step_link_dict)
+    reference_keyid = key_link_dict.keys()[0]
+    reference_link = key_link_dict[reference_key]
 
-    # iterate over the remaining links to compare their properties
+    # Iterate over all links to check their signature and
+    # compare their properties with a reference_link
     for keyid, link in six.iteritems(key_link_dict):
       keys_dict = {keyid: layout.keys.get(keyid)}
 
       # check if the authorized functionary has signed the link
       try:
         verify_link_signatures(link, keys_dict)
       except SignatureVerificationError as e:
-        raise ThresholdVerificationError("Link not signed by"
-            " authorized functionary!")
+        raise ThresholdVerificationError("Link '{0}' not signed by"
```

Comment:
Actually we already check in [verify_all_steps_signatures](https://github.com/in-toto/in-toto/blob/output-enh/in_toto/verifylib.py#L251-L254) if a signature is correct and the functionary key was authorized to sign.

Here we check again (by filename keyid) to prevent threshold circumvention by duplicating a validly signed link and changing one filename.